python/grass/temporal/univar_statistics.py

```diff
@@ -57,7 +57,7 @@ def compute_univar_stats(registered_map_info, stats_module, fs, rast_region=Fals
     )
 
     stats_module.inputs.map = id
-    if rast_region:
+    if rast_region and (stats_module.inputs.zones or stats_module.name == "r3.univar"):
```

Comment:
This is coherent with your explanations in https://github.com/OSGeo/grass/pull/4577#issuecomment-2433687422

But now, the docstring of the `rast_region` parameter of this function (`compute_univar_stats()`) should be adjusted to reflect the change.

Currently, it reads:
```
    :param rast_region: If set True ignore the current region settings
           and use the raster map regions for univar statistical calculation.
           Only available for strds.
```